### PR TITLE
[CMDBUF] Implement kernel binary update for L0 adapter

### PR DESCRIPTION
**Pull request by Bensuo**
- Implement kernel binary update in L0 adapter
- Platform and driver query support for new extension functions
- Update command buffer documentation to specify arg requirements for binary update

### CODE REVIEW

**Comment by Bensuo** [comment]
DPC++ PR: https://github.com/intel/llvm/pull/16154

File: source/adapters/level_zero/platform.cpp [hunk 1]
```diff
@@ -320,6 +320,22 @@ ur_result_t ur_platform_handle_t_::initialize() {
                 &ZeMutableCmdListExt
                      .zexCommandListUpdateMutableCommandWaitEventsExp))) == 0);
 
+  ZeMutableCmdListExt.Supported &=
```
Comment (EwanC) [inline source/adapters/level_zero/platform.cpp:323]: Noting that this conflicts with https://github.com/oneapi-src/unified-runtime/pull/2351, not that I think there's any action to do yet but address conflicts.

**Review by EwanC — APPROVED** [review]

File: scripts/core/EXP-COMMAND-BUFFER.rst [hunk 2]
```diff
@@ -256,6 +256,11 @@ ${x}CommandBufferAppendKernelLaunchExp. The command can then be updated
 to use the new kernel handle by passing it to
 ${x}CommandBufferUpdateKernelLaunchExp.
 
+.. important::
+    When updating the kernel handle of a command you must also provide all
```
Comment (reble) [inline scripts/core/EXP-COMMAND-BUFFER.rst:260]: Can we use passive voice here, and avoid using you?
Might better align with the existing documentation.

**Comment by EwanC** [comment]
@oneapi-src/unified-runtime-level-zero-write Could you review this please

**Review by pbalcer — APPROVED** [review]
l0 lgtm, but we should merge https://github.com/oneapi-src/unified-runtime/pull/2351 first.

**Pull request merged**